gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -1696,3 +1742,101 @@ MM_MemoryPoolAddressOrderedList::releaseFreeMemoryPages(MM_EnvironmentBase* env)
 	return releasedBytes;
 }
 #endif
+
+MMINLINE uintptr_t
+MM_MemoryPoolAddressOrderedList::getAlignedSize(MM_EnvironmentBase *env, void *addrBase, void *addrTop)
+{
+	Assert_MM_true((uintptr_t)addrTop >= ((uintptr_t)addrBase + _minimumFreeEntrySize));
+	return ((uintptr_t)MM_CardTable::alignWithCard(addrTop, true, CARD_SIZE) - (uintptr_t)MM_CardTable::alignWithCard(addrBase, false, CARD_SIZE));
+}
+
+MM_HeapLinkedFreeHeader *
+MM_MemoryPoolAddressOrderedList::doFreeEntryAlignmentUpTo(MM_EnvironmentBase *env, MM_HeapLinkedFreeHeader *freeEntry)
+{
+	MM_HeapLinkedFreeHeader *ret = NULL;
+	bool const compressed = env->compressObjectReferences();
+	MM_HeapLinkedFreeHeader *currentFreeEntry = _firstUnAlignmentFreeEntry;
+	MM_HeapLinkedFreeHeader *previousFreeEntry = _preFirstUnAlignmentFreeEntry;
+
+	MM_HeapLinkedFreeHeader *nextFreeEntry = NULL;
+	void* newStartFreeEntry = NULL;
+	void* endFreeEntry = NULL;
+	void* newEndFreeEntry = NULL;
+	UDATA freeEntrySize = 0;
+	UDATA lostToAlignment = 0;
+
+	UDATA freeBytes = _freeMemorySize;
+	UDATA freeEntryCount = _freeEntryCount;
+	if ((_heapFreeList->getSize() < _minimumFreeEntrySize) || (currentFreeEntry->getSize() < _minimumFreeEntrySize)) {
+		OMRPORT_ACCESS_FROM_ENVIRONMENT(env);
+		omrtty_printf("doFreeEntryAlignmentUpTo start memoryPool=%p, currentFreeEntry=%p, previousFreeEntry=%p, _heapFreeList=%p\n",
+				this, currentFreeEntry, previousFreeEntry, _heapFreeList);
+		Assert_MM_true(false);
+	}
+	while ((NULL != currentFreeEntry) && ((uintptr_t)currentFreeEntry <= (uintptr_t)freeEntry)) {
+		freeEntrySize = currentFreeEntry->getSize();
+		endFreeEntry = (void *) ((UDATA)currentFreeEntry + freeEntrySize);
+		newStartFreeEntry = MM_CardTable::alignWithCard((void *) currentFreeEntry, false, CARD_SIZE);
+		newEndFreeEntry = MM_CardTable::alignWithCard((void *) endFreeEntry, true, CARD_SIZE);
+		nextFreeEntry = currentFreeEntry->getNext(compressed);
+
+		if (((UDATA)currentFreeEntry != (UDATA)newStartFreeEntry) || ((UDATA)endFreeEntry != (UDATA)newEndFreeEntry)) {
```

Comment:
I believe, the rest of AOL code casts to (uint8_t *) when doing various comparisons. Or what ever it is, let's just be consistent (it should not be UDATA since OMR is almost completely free of it).